docs/core_docs/docs/integrations/vectorstores/couchbase.mdx

```diff
@@ -0,0 +1,367 @@
+---
+hide_table_of_contents: true
+sidebar_class_name: node-only
+---
+
+import CodeBlock from "@theme/CodeBlock";
+
+# Couchbase
+
+[Couchbase](http://couchbase.com/) is an award-winning distributed NoSQL cloud database that delivers unmatched versatility, performance, scalability, and financial value for all of your cloud, mobile,
+AI, and edge computing applications. Couchbase embraces AI with coding assistance for developers and vector search for their applications.
+
+Vector Search is a part of the [Full Text Search Service](https://docs.couchbase.com/server/current/learn/services-and-indexes/services/search-service.html) (Search Service) in Couchbase.
+
+This tutorial explains how to use Vector Search in Couchbase. You can work with both [Couchbase Capella](https://www.couchbase.com/products/capella/) and your self-managed Couchbase Server.
+
+## Installation
+
+You will need couchbase and langchain community to use couchbase vector store. For this tutorial, we will use OpenAI embeddings
+
+```bash npm2yarn
+npm install couchbase @langchain/openai @langchain/community
+```
+
+## Create Couchbase Connection Object
+
+We create a connection to the Couchbase cluster initially and then pass the cluster object to the Vector Store. Here, we are connecting using the username and password.
+You can also connect using any other supported way to your cluster.
+
+For more information on connecting to the Couchbase cluster, please check the [Node SDK documentation](https://docs.couchbase.com/nodejs-sdk/current/hello-world/start-using-sdk.html#connect).
+
+```typescript
+import { Cluster } from "couchbase";
+
+const connectionString = "couchbase://localhost"; // valid couchbase connection string
```

Comment:
Can you add `couchbases://localhost` for TLS in the comments as well? This was an ask from PM